### PR TITLE
chore: deprecate `VRA7_` environmental variables

### LINKED ISSUE #518: References to `VRA7_` need to be updated
### Code of Conduct

This project has a Code of Conduct that all participants are expected to understand and follow:
* [ ] I have read and agree to the project's [Code of Conduct](https://github.com/vmware/terraform-provider-vra/blob/main/CODE-OF-CONDUCT.md)

### vRA Version
8.16.0.33697

### Terraform Version
1.5.2

### vRA Terraform Provider Version
0.8.7

### Affected Resource(s)
N/A

### Terraform Configuration Files
N/A

```hcl
# Copy-paste your Terraform configurations here.
# For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file.
```

### Expected Behavior
Expect that provider environment variables VRA7_INSECURE and VRA7_REAUTHORIZE_TIMEOUT be renamed to VRA_INSECURE and VRA_REAUTHORIZE_TIMEOUT.

### Actual Behavior
Environment variables have not been updated since VRA7.

### References
https://github.com/vmware/terraform-provider-vra/blob/main/vra/provider.go#L35
https://github.com/vmware/terraform-provider-vra/blob/main/vra/provider.go#L41
https://github.com/vmware/terraform-provider-vra/blob/7dab4873ffa6cee62b646f7cf6ffa55c5aee6da7/website/docs/index.html.markdown?plain=1#L69

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by tenthirtyam**
### Description

- Updates `reauthorize_timeout` to accept `VRA_INSECURE`, and `VRA7_INSECURE`. Use of `VRA7_INSECURE` will trigger the depreciation notice.
- Updates `insecure` to accept `VRA_REAUTHORIZE_TIMEOUT`, and `VRA7_REAUTHORIZE_TIMEOUT`. Use of `VRA7_REAUTHORIZE_TIMEOUT` will trigger the depreciation notice.
- Updates documentation.

### Reference

Closes #518

### CODE REVIEW

File: vra/provider.go [hunk 1]
```diff
@@ -32,15 +34,17 @@ func Provider() *schema.Provider {
 			},
 			"insecure": {
 				Type:        schema.TypeBool,
-				DefaultFunc: schema.EnvDefaultFunc("VRA7_INSECURE", nil),
+				DefaultFunc: envDefaultFunc("VRA_INSECURE", "VRA7_INSECURE"),
 				Optional:    true,
+				Deprecated:  "The 'VRA7_INSECURE' environment variable is deprecated. Please use 'VRA_INSECURE' instead.",
```
[the remark this answers is not in the document]
Comment (tenthirtyam) [reply vra/provider.go:39]: Right! Great point!

File: vra/provider.go [hunk 2]
```diff
@@ -32,15 +34,17 @@ func Provider() *schema.Provider {
...
 				Description: "Specify whether to validate TLS certificates.",
 			},
 			"reauthorize_timeout": {
 				Type:        schema.TypeString,
-				DefaultFunc: schema.EnvDefaultFunc("VRA7_REAUTHORIZE_TIMEOUT", nil),
+				DefaultFunc: envDefaultFunc("VRA_REAUTHORIZE_TIMEOUT", "VRA7_REAUTHORIZE_TIMEOUT"),
```
[the remark this answers is not in the document]
Comment (tenthirtyam) [reply vra/provider.go:44]: Something like;


```go
insecure := {
	Type:        schema.TypeBool,
	DefaultFunc: schema.MultiEnvDefaultFunc([]string{"VRA_INSECURE", "VRA7_INSECURE"}, nil),
	Optional:    true,
	Description: "Specify whether to validate TLS certificates.",
	ValidateDiagFunc: func(val interface{}, path cty.Path) diag.Diagnostics {
		var diags diag.Diagnostics
		if envVar, ok := os.LookupEnv("VRA7_INSECURE"); ok && envVar != "" {
			diags = append(diags, diag.Diagnostic{
				Severity: diag.Warning,
				Summary:  "Deprecated environment variable",
				Detail:   "The 'VRA7_INSECURE' environment variable is deprecated. Please use 'VRA_INSECURE' instead.",
			})
		}
		return diags
	},
}

```

[the remark this answers is not in the document]
Comment (frodenas) [reply vra/provider.go:44]: Exactly! :)

[the remark this answers is not in the document]
Comment (tenthirtyam) [reply vra/provider.go:54]: Resolved in 4d52099.

**Comment by tenthirtyam** [comment]
@frodenas - still need to test.

**Review by frodenas — APPROVED** [review]
I successfully tested these changes on my local environment.

**Comment by tenthirtyam** [comment]
Thanks!!

**Pull request merged**